Programs/_testembed.c

```diff
@@ -1550,6 +1550,46 @@ static int test_init_setpythonhome(void)
 }
 
 
+static int test_init_is_python_build(void)
+{
+    char *env = getenv("TESTHOME");
+    if (!env) {
+        error("missing TESTHOME env var");
+        return 1;
+    }
+    wchar_t *home = Py_DecodeLocale(env, NULL);
+    if (home == NULL) {
+        error("failed to decode TESTHOME");
+        return 1;
+    }
+
+    PyConfig config;
+    _PyConfig_InitCompatConfig(&config);
+    config_set_program_name(&config);
+    // Ensure 'home_was_set' is turned on in getpath.py
+    config_set_string(&config, &config.home, home);
+    PyMem_RawFree(home);
+    putenv("TESTHOME=");
+
+    config._is_python_build = INT_MAX;
+    env = getenv("NEGATIVE_ISPYTHONBUILD");
+    if (env) {
+        if (strcmp(env, "0") != 0) {
+            config._is_python_build++;
+        }
+        putenv("NEGATIVE_ISPYTHONBUILD=");
```

Comment:
This shouldn't be necessary, right? Nobody else reads this, and we aren't going to loop back through here.